behave/features/steps/diagnosis_debt_court_proceedings.py

```diff
@@ -1,26 +1,67 @@
-    context.helperfunc.driver().switch_to.active_element.send_keys(CLA_USER_TO_ASSIGN_CASES_TO)
-    # check that there are results and click on top value in dropdown
-    # this element is not displayed until the user has typed in the search box
-    # choose the first in the list and check the name of the person matches
-    span_x_path = "//ul[@class='select2-results']/li/div/span"
-    li_x_path = "//ul[@class='select2-results']/li"
-    assert CLA_USER_TO_ASSIGN_CASES_TO == context.helperfunc.find_many_by_xpath(span_x_path)[0].text,  \
-        f"Failed with {CLA_USER_TO_ASSIGN_CASES_TO} and {context.helperfunc.find_many_by_xpath(span_x_path)[0].text}"
-    # select the first in the list
-    context.helperfunc.find_many_by_xpath(li_x_path)[0].click()
+    select_value_from_list(context, label="Search for existing user", op="startswith", value=CLA_USER_TO_ASSIGN_CASES_TO)
     # there will be an alert asking if you wish to continue
     assert context.helperfunc.driver().switch_to.alert is not None, "No alert confirming you want to add the user"
     context.helperfunc.driver().switch_to.alert.accept()
 
 
+@then(u'I select the "{category}" knowledge base category')
+def step_impl(context, category):
+    select_value_from_list(context, label="Law category", value=category)
+    # Need to wait for a bit for the ajax event to complete before continuing to the next step
+    time.sleep(2)
+
+
+@then(u'I select the alternative help organisations "{organisation}"')
+def step_impl(context, organisation):
+    name, _ = organisation.split(' - ')
+    search_input = context.helperfunc.find_by_xpath(f"//input[@placeholder='Search providers and other help organisations']")
+    search_input.clear()
+    search_input.send_keys(name)
+    submit = search_input.find_element_by_xpath('following-sibling::*')
+    submit.click()
+    # Need to wait for a bit for the ajax event to complete before continuing to the next step
+    time.sleep(1)
+    search_results_form = context.helperfunc.find_by_xpath('//form[@name="alternative_help"]')
+    # This will only find the first search result which is fine because we are searching for a specific organisation
+    parent_wrapper = search_results_form.find_element_by_xpath('.//input[@name="selected_providers"]/ancestor::div[1]')
+    assert parent_wrapper.find_element_by_css_selector('.FormRow-label strong').text == organisation
+    parent_wrapper.find_element_by_css_selector('label.FormRow-label').click()
+
+
+@then(u'I enter "{comment}" in the Assignment comments box')
+def step_impl(context, comment):
+    text_area = context.helperfunc.find_by_xpath('//textarea[@name="assign-notes"][@placeholder="Assignment comments"]')
+    text_area.send_keys(comment)
+    assert text_area.get_attribute("value") == comment
+
+
+@then(u'I select Assign alternative help')
+def step_impl(context):
+    submit_btn = context.helperfunc.find_by_xpath('//button[@name="assign-alternative-help"]')
```

Comment:
Is this not a hyperlink rather than a button? Please can we discuss - I have the same step coming up (so either way should go to common_steps).